Dockerfile

```diff
@@ -0,0 +1,64 @@
+###############
+# Build stage #
+###############
+
+FROM  elixir:1.5-alpine
+
+WORKDIR /build
+
+# We need hex and rebar build tools installed.
+RUN mix local.hex --force && mix local.rebar --force
+
+# Copy all the source files. We can't just use `COPY . .` because that would
+# also copy deps, _build, and various other things that would cause trouble
+# with the build. Also, `COPY dir .` will copy the contents of the dir instead
+# of the dir itself, so we have to manually copy each dir into its own
+# destination.
+COPY mix.* ./
+COPY config ./config
+COPY lib ./lib
+COPY test ./test
+COPY rel ./rel
+
+# Build the release to put in the next image.
+ENV MIX_ENV=prod
+RUN mix deps.get
+RUN mix compile
+RUN mix release --env=prod
+
+
+###############
+# Image stage #
+###############
+
+# We don't actually need Erlang/Elixir installed, because it's all included in
+# the release package. Thus, we start from the base alpine image.
+FROM alpine:3.7
```

Comment:
The build image seems to be based on Alpine 3.6. It should produce forward-compatible things, although I don't understand much about what distillery is doing yet.

Reply:
We're building "deployments" which only need basic OS stuff -- the right Erlang/Elixir/whatever is shipped along with that. If we exclude the Erlang bits, we need to use a container that has the same Erlang release as we were built with, otherwise as far as I can tell pretty much anything goes.